app/users/index/controller.js

```diff
@@ -1,106 +1,101 @@
-    defaultValue: "username",
-    replace: true,
-    refresh: true,
-  },
-});
-
-const UsersIndexController = Controller.extend(UsersQueryParams.Mixin, {
-  session: service("session"),
-  router: service("router"),
-
-  currentUser: reads("session.data.user"),
-
-  viewUserProfile: action(function (userId) {
-    return this.router.transitionTo("users.edit", userId);
-  }),
-
-  selectedSupervisor: computed(
-    "prefetchData.lastSuccessful.value.supervisor",
-    "store",
-    "supervisor",
-    function () {
-      return this.supervisor && this.store.peekRecord("user", this.supervisor);
-    }
-  ),
-
-  setup() {
+export default class UsersIndexController extends Controller {
+  queryParams = ["search", "supervisor", "active", "ordering"];
+  @service session;
+  @service router;
+  @service store;
+  @tracked currentUser = this.session.data.user;
+  @tracked search = "";
+  @tracked supervisor = null;
+  @tracked active = "1";
+  @tracked ordering = "username";
+
+  constructor(...args) {
+    super(...args);
     this.prefetchData.perform();
-    this.data.perform();
-  },
-
-  reset() {
-    /* istanbul ignore next */
-    this.resetQueryParams();
-  },
+  }
+
+  _fetchData = trackedTask(this, this.data, () => {
+    return [this.supervisor, this.search, this.ordering, this.active];
+  });
+
+  get selectedSupervisor() {
+    return this.supervisor && this.store.peekRecord("user", this.supervisor);
+  }
+
+  get fetchData() {
+    return this._fetchData ?? {};
+  }
+
+  get allQueryParams() {
+    return {
+      supervisor: this.supervisor,
+      search: this.search,
+      active: this.active,
+      ordering: this.ordering,
+    };
+  }
+
+  resetQueryParams() {
+    this.search = "";
+    this.supervisor = null;
+    this.active = "1";
+    this.ordering = "username";
+  }
+
+  @restartableTask
+  *resetFilter() {
+    yield this.resetQueryParams();
+  }
 
-  queryParamsDidChange({ shouldRefresh }) {
-    if (shouldRefresh) {
-      this.data.perform();
-    }
-  },
+  @action
+  viewUserProfile(userId) {
+    return this.router.transitionTo("users.edit", userId);
+  }
 
-  prefetchData: task(function* () {
+  @restartableTask
+  *prefetchData() {
     const supervisorId = this.supervisor;
 
     return yield hash({
       supervisor: supervisorId && this.store.findRecord("user", supervisorId),
     });
-  }).restartable(),
+  }
 
-  data: task(function* () {
+  @restartableTask
+  *data() {
+    yield Promise.resolve();
     const date = moment().format("YYYY-MM-DD");
 
     yield this.store.query("employment", { date });
     yield this.store.query("worktime-balance", { date });
 
     return yield this.store.query("user", {
       ...this.allQueryParams,
-      ...(this.get("currentUser.isSuperuser")
+      ...(this.currentUser.isSuperuser
         ? {}
         : {
-            supervisor: this.get("currentUser.id"),
+            supervisor: this.currentUser.id,
           }),
     });
-  }).restartable(),
+  }
```

Comment:
We could implement this with the route (model) and the native query params, too. Or use 
 [ember-async-data](https://github.com/tracked-tools/ember-async-data). This whole controller feels so overcomplicated. :man_shrugging: